### PR TITLE
Simple parsing for textstyles and emojis

### PR DESCRIPTION
**Pull request by erinyoo**
Hey! So I've created the basic stylization/parsing when the message is sent.

What's added:
New packages (stylization.text & stylization.emojis)
 - EmojiUtils.java (emoji parsing)
 - TextUtils.java (text parsing/stylization)

Edited
 - ChatServlet.java (added parsing when sending messages)
 - chat.jsp (a little info about how to stylize)

### CODE REVIEW

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 1]
```diff
@@ -87,6 +89,20 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
+        <p>😀- :grinning:  😍- :heart_eyes:   😂- :laughing:   😭- :crying:   😉- :winking:</p>
+        <p>😎- :grinning:   😞- :disappointed:   😕- :confused:   😐- :neutral:  😠- :angry:</p>
+    </div>
+
+    <h2>Text Style Shortcuts</h2>
+    <div id="shortcuts">
+        <p>Strikethrough: =STYLEDTEXT=</p>
+        <p>Bold: *STYLEDTEXT*</p>
+        <p>Italics: ~STYLEDTEXT~</p>
+        <p>Underine: /STYLEDTEXT/</p>
```
Comment (tucosh) [inline src/main/webapp/WEB-INF/view/chat.jsp]: Underline

File: src/main/java/codeu/controller/ChatServlet.java [hunk 2]
```diff
@@ -20,14 +20,22 @@
 import codeu.model.store.basic.ConversationStore;
 import codeu.model.store.basic.MessageStore;
 import codeu.model.store.basic.UserStore;
+
+import java.awt.*;
```
Comment (tucosh) [inline src/main/java/codeu/controller/ChatServlet.java]: This and other import statements are not needed, remove. If you do require an import, be sure to specify the fully qualified class name, rather than importing an entire package. (Interview tip!)

File: src/main/java/codeu/stylization/emojis/EmojiUtils.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package codeu.stylization.emojis;
+
+public class EmojiUtils {
+
+    /** The message that has been parsed for emojis */
+    private String messageParsed;
+
+    /** The original message that was passed (unparsed for emojis) */
+    private String originalMessage;
```
Comment (tucosh) [inline src/main/java/codeu/stylization/emojis/EmojiUtils.java]: Make originalMessage final if it cannot be changed (interview tip..)

File: src/main/java/codeu/stylization/emojis/EmojiUtils.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        parseEmojis();
+        return messageParsed;
+    }
+
+    public String getOriginalMessage() {
+        return originalMessage;
+    }
+
+    private void parseEmojis() {
+        String messageUnparsed = originalMessage;
```
Comment (tucosh) [inline src/main/java/codeu/stylization/emojis/EmojiUtils.java]: Just chain all these together into a single statement:

messageUnparsed = originalMessage
  .replaceAll(":grinning:", "\uD83D\uDE00")
  .replaceAll(":laughing:", "\uD83D\uDE02")
  ..etc

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 5]
```diff
@@ -13,6 +13,7 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 --%>
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8"%>
```
Comment (tucosh) [inline src/main/webapp/WEB-INF/view/chat.jsp]: Is this line needed?

File: src/main/java/codeu/stylization/text/TextUtils.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+    /**
+     * Function will return -1, -1 if none of the stylizations encountered is correctly formatted.
+     * Otherwise, it will return the indices in which it encounters a correctly formatted stylization.
+     * @param stylization The type of stylization the method should look for
+     * @param messageUnparsed The message that is currently unparsed for that particular stylization
+     * @return An integer array that returns the indicies in which the stylization first appears (includes
+     * the style indicators (*, /, ~, =)
+     */
+    private int[] getIndexOfStyle(String stylization, String messageUnparsed) {
```
Comment (tucosh) [inline src/main/java/codeu/stylization/text/TextUtils.java]: This is pretty difficult to follow. Really you need to add a unit test TextUtilsTest that both tests and illustrates what this does, that would make it considerably easier for someone examining this class to understand what it does. String manipulation is tricky (and therefore popular in interviews..) and there are always corner cases so a thorough slate of tests is definitely indicated

File: src/main/java/codeu/stylization/text/TextUtils.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+            styleEnding = "</i>";
+        }
+        if (stylization.equals("strikethrough")) {
+            style = "<strike>";
+            styleEnding = "</strike>";
+        }
+        StringBuilder messageReplaced = new StringBuilder();
+        int[] indicies = getIndexOfStyle(stylization, messageUnparsed);
+        if(indicies[0] == -1) return messageUnparsed;
+        while (messageUnparsed.length() != 0 && indicies[0] != -1) {
```
Comment (tucosh) [inline src/main/java/codeu/stylization/text/TextUtils.java]: needs a few comments explaining the logic

**Comment by tucosh** [comment]
Seems to work.. good job!
Just a few suggestions
